sdks/unity/Tests/Runtime/PlayMode/AgonesSdkIntegrationTests.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2024 Google LLC
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Agones;
+using System.Threading.Tasks;
+using System;
+
+namespace Tests.Runtime.Agones
+{
+    public class AgonesSdkIntegrationTests
+    {
+
+        private AgonesSdk sdk;
+        private AgonesBetaSdk betaSdk;
+        private AgonesAlphaSdk alphaSdk;
+
+        [UnitySetUp]
+        public IEnumerator UnitySetUp()
+        {
+            GameObject gameObject = new GameObject();
+            gameObject.AddComponent<AgonesSdk>();
+            gameObject.AddComponent<AgonesAlphaSdk>();
+            gameObject.AddComponent<AgonesBetaSdk>();
+            yield return null;
+
+            this.sdk = gameObject.GetComponent<AgonesSdk>();
+            this.betaSdk = gameObject.GetComponent<AgonesBetaSdk>();
+            this.alphaSdk = gameObject.GetComponent<AgonesAlphaSdk>();
+
+            Assert.IsNotNull(this.sdk);
+            Assert.IsNotNull(this.betaSdk);
+            Assert.IsNotNull(this.alphaSdk);
+        }
+
+        [UnityTest]
+        public IEnumerator TestSdk()
+        {
+            var task = RunSdkTests();
```

Comment:
This is an interesting way of setting up a test. I would have expected more test methods each with a smaller amount of test assertions (e.g. "I can add to a Agones list" separated from "I can remove from an Agones list" as a simplistic example).

The way this is set up the AgonesSDK and AgonesBetaSDK are tested sequentially, started from this one UnityTest method. I wouldn't expect a test to be set up like this. Just curious what some of the pros/cons of the current set up you would suggest this set up has?

Reply:
I simply copied the structure from the node SDK here: https://github.com/googleforgames/agones/blob/main/test/sdk/nodejs/testSDKClient.js